service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/update/UpdateControlledResourceMetadataStep.java

```diff
@@ -0,0 +1,66 @@
+package bio.terra.workspace.service.resource.controlled.flight.update;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.resource.controlled.ControlledResource;
+import bio.terra.workspace.service.resource.controlled.ControlledResourceMetadataManager;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import java.util.UUID;
+
+public class UpdateControlledResourceMetadataStep implements Step {
+
+  private final ResourceDao resourceDao;
+  private final UUID resourceId;
+  private final UUID workspaceId;
+  private final ControlledResourceMetadataManager controlledResourceMetadataManager;
+
+  public UpdateControlledResourceMetadataStep(
+      ControlledResourceMetadataManager controlledResourceMetadataManager,
+      ResourceDao resourceDao,
+      UUID workspaceId,
+      UUID resourceId) {
+    this.resourceDao = resourceDao;
+    this.resourceId = resourceId;
+    this.workspaceId = workspaceId;
+    this.controlledResourceMetadataManager = controlledResourceMetadataManager;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    final FlightMap inputParameters = flightContext.getInputParameters();
+    final ControlledResource resource =
+        inputParameters.get(JobMapKeys.REQUEST.getKeyName(), ControlledResource.class);
+    final String resourceName =
+        inputParameters.get(ControlledResourceKeys.RESOURCE_NAME, String.class);
+    final String resourceDescription =
+        inputParameters.get(ControlledResourceKeys.RESOURCE_DESCRIPTION, String.class);
+    final AuthenticatedUserRequest userRequest =
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    controlledResourceMetadataManager.updateControlledResourceMetadata(
+        workspaceId, resourceId, resourceName, resourceDescription, userRequest);
+    return StepResult.getStepResultSuccess();
+  }
+
+  /**
+   * Restore the previous values of the metadata.
+   *
+   * @param flightContext
+   */
+  @Override
+  public StepResult undoStep(FlightContext flightContext) throws InterruptedException {
+    final FlightMap workingMap = flightContext.getWorkingMap();
+    final String previousName =
+        workingMap.get(ControlledResourceKeys.PREVIOUS_RESOURCE_NAME, String.class);
+    final String previousDescription =
+        workingMap.get(ControlledResourceKeys.PREVIOUS_RESOURCE_DESCRIPTION, String.class);
+    resourceDao.updateResource(workspaceId, resourceId, previousName, previousDescription);
```

Comment:
Since we don't do any locking, it is possible that the resource was deleted between the time this flight started and when this executes this step. You would get a not-found raised here. I think that should be tolerated and not cause the flight to fail dismally.

Reply:
Yes, you are right